lib/dataset/variable_instantiate_bin_elements.cpp

```diff
@@ -55,18 +55,19 @@ dict_to_compact_string(const scipp::dataset::SizedDict<Key, Value> &dict,
 
 template <>
 std::string Formatter<core::bin<DataArray>>::format(const Variable &var) const {
-  const auto &buffer = var.bin_buffer<DataArray>();
-  std::string margin(8, ' ');
+  const auto &[indices, dim, buffer] = var.constituents<DataArray>();
+  std::string margin(10, ' ');
```

Comment:
I think we should limit the line length a little more (in `dict_to_compact_string`), as this no longer wraps so well in the html repr?
![Screenshot at 2023-01-24 10-58-17](https://user-images.githubusercontent.com/39047984/214263146-a5a00e2d-98c7-49ee-a531-d870f5d3b62b.png)

Or have a better rule for when to wrap? I think currently I'm checking if the line is too long after having added to the line, while we should probably check before appending whether the new addition will cause the line to overflow and break before appending?